### DISCUSSION TITLE
Question about the problem: [Killed python3 -m bot]

### DISCUSSION THREAD

**Discussion by Namlee123** [anasty17/mirror-leech-telegram-bot#1403] [General]
Hello Anasty17!
I have a problem, need your advice.
Once I deployed the bot and the bot worked for a few hours.
then in the log appears the line:
"start.sh: line 1:16 Killed python3 -m bot"
Then my bot restarted.

I tried again a few times, the bot works normally, but when the log shows the line "killed python" as above, the bot restarts.
Hope you can give me some ideas to fix this problem.
Thank you.

**Comment by anasty17** [comment]
Machine specification?

**Reply by Namlee123** [reply]
I use:
Ram: 8 GB,
CPU: [AMD Ryzen 7] - 8core/16t
Cache: 16 MB

But do I ever run out of cache?
I'm testing python's killer bug, some people on the forum recommend me:
import gc
gc.set_threshold(0)

I have a similar issue. I have a python script which runs for a few minutes creating matplotlib plot files. If i run the script from the commandline directly via python script.py then everything work fine. As soon as i run this from a bash script, the memory usage increases over time an finaly stops with the same kill message from above. I haven't yet found a solution.

**Comment by Namlee123** [comment]
I also switched to testing your repo on the Render platform's Web server (512M Ram, 0.1 CPU) , Railway (512M Ram, 2 vcpu) in the free tier.
All thing implementations work fine. but after more than 8h running:
Render is still normal.
Railway [Killed python3 -m bot]

**Reply by anasty17** [reply]
Is it vps!!!! If not vps close this

**Reply by Namlee123** [reply]
I run the repo on Vps.
I also test on my personal computer and web server to see if they are killed python or not?
Result (Killer error) on all 3 platforms.
Just hope can find the cause and solution on Vps because I mainly use vps.

**Reply by anasty17** [reply]
Docker ?

**Comment by anasty17** [comment]
I have used it on vps and on my local computer and nothing wrong

**Reply by Namlee123** [reply]
I had this problem when downloading some large torrents (100G) with high download speed (40MB). Small, low-speed torrents are virtually unaffected.
It randomly restarts when it loads for a few hours or tens of minutes,
it does not have a fixed time for the error to arise.
![Capture](https://user-images.githubusercontent.com/92989894/226176114-24f0595d-5420-426f-98a0-def4c32a3184.PNG)
